Add a CMake based compilation option

Yuki-cpp opened the pull request:
Adds a CMakeList.txt to make build the project via CMake. Also includes a dummy hello world program to test on. As in CMake the install command is dedicated to installing the produced software, renamed the install command from the Makefile to install_deps

This also updates the README to provide guidance on how to compile with
 Cmake.

Commit:
Add a CMake based compilation option

Adds a CMakeList.txt to make build the project via CMake. Also includes
a dummy hello world program to test on. As in CMake the install command
is dedicated to installing the produced software, renamed the install
command from the Makefile to install_deps

This also updates the README to provide guidance on how to compile with
 Cmake.

rektpwner commented on Makefile:
```diff
@@ -13,8 +13,8 @@ all: install default
-.PHONY: install
-install:
+.PHONY: install_deps
+install_deps:
```
Since `install` is renamed to `install_deps`, please change this part:
```
.PHONY: all
all: install default
```
You can rename or remove `install` label, for example.

Commit:
Set flags parity between CMake and Makefile

Added the same warning flags in the Makefile that were added to the Cmake based compilation

Commit:
Removed the capstone not found message

It was dead code as not finding capstone would stop the process before the message
